pkg/controller/traceflow/controller.go

```diff
@@ -277,20 +303,21 @@ func (c *Controller) checkTraceflowStatus(tf *opsv1alpha1.Traceflow) error {
 		}
 	}
 	if sender && receiver {
+		c.deallocateTagForTF(tf)
 		return c.updateTraceflowStatus(tf, opsv1alpha1.Succeeded, "", 0)
 	}
-	if time.Now().UTC().Sub(tf.CreationTimestamp.UTC()).Seconds() > timeout {
-		return c.updateTraceflowStatus(tf, opsv1alpha1.Failed, "traceflow timeout", 0)
+	// CreationTimestamp is of second accuracy.
+	if time.Now().Unix() > tf.CreationTimestamp.Unix()+int64(timeoutDuration.Seconds()) {
```

Comment:
`timeoutDuration.Seconds()` is still seconds, this could lead to very fast failure, didn't it fail tests?

Reply:
Sorry, false comment, I mistook "Unix".

Reply:
Did not get what you mean. timeoutDuration is 120 seconds. Why it leads to fast failure?